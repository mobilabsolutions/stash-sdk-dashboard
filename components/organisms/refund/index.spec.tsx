import { testRender } from '../../../test_utils'
// import { shallow, mount } from 'enzyme'
import RefundForm from './index'

it('RefundForm Should Render', () =>
  testRender(RefundForm, {
    initialRefund: 12345,
    currencyId: 'EUR'
  }))

it('RefundForm Should use loading status', () =>
  testRender(RefundForm, {
    initialRefund: 12345,
    isLoading: true,
    currencyId: 'EUR'
  }))
// it('RefundForm Should disable amount initially', () => {
//   const form = mount(<RefundForm initialRefund={12345} currencyId={'EUR'} />)
//   const RefundInput = form.find('[name="refund"]').get(0)
//   expect(RefundInput.props.disabled).toBe(true)
// })

// it('RefundForm Should disable amount only when full refund is picked', () => {
//   const form = mount(<RefundForm initialRefund={12345} currencyId={'EUR'} />)
//   const RefundInput = form.find('[name="refund"]').get(0)
//   expect(RefundInput.props.disabled).toBe(true)
// })
