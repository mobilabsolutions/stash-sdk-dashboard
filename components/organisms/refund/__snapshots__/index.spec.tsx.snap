// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefundForm Should Render 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="refund__ContentContainer-a7ak7z-3 icpIBn"
  >
    <div
      className="refund__FormItemContainer-a7ak7z-1 iBWkWY"
    >
      <span
        className="refund__Value-a7ak7z-0 iETVAo"
      >
        <label
          className="radio__Label-sc-1hx8hak-0 kZePut"
          label="Partial"
        >
          Partial
          <input
            checked={false}
            name="refundType"
            onChange={[Function]}
            type="radio"
            value="partial"
          />
          <span />
        </label>
        <label
          className="radio__Label-sc-1hx8hak-0 kZePut"
          label="Full"
        >
          Full
          <input
            checked={true}
            name="refundType"
            onChange={[Function]}
            type="radio"
            value="full"
          />
          <span />
        </label>
      </span>
    </div>
    <div
      className="field_wrapper-n32k2e-0 irXinp"
      onClick={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <div
        style={
          Object {
            "minWidth": 100,
          }
        }
      >
        <h3
          className="h3-sc-1przhcr-0 bqBPGG"
        >
          Refund
        </h3>
      </div>
      <div
        className="input__InputErrorWrapper-sc-1dz7i8-0 cAUOWl"
      >
        <div
          className="input_wrapper-sc-15hmhfy-0 duFQAu"
          style={
            Object {
              "marginLeft": "16px",
              "width": "100%",
            }
          }
        >
          <input
            autoFocus={false}
            className="input_currency__HtmlInput-obstnl-0 cVRLQV"
            disabled={true}
            name="refund"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            placeholder="Refund"
            type="text"
            value="€12,345.00"
          />
        </div>
      </div>
    </div>
    <div
      className="field_wrapper-n32k2e-0 irXinp"
      onClick={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <div
        style={
          Object {
            "minWidth": 100,
          }
        }
      >
        <h3
          className="h3-sc-1przhcr-0 bqBPGG"
        >
          Description
        </h3>
      </div>
      <div
        className="input__InputErrorWrapper-sc-1dz7i8-0 cAUOWl"
      >
        <div
          className="input_wrapper-sc-15hmhfy-0 hwqBS"
          style={
            Object {
              "marginLeft": "16px",
              "width": "100%",
            }
          }
        >
          <input
            autoFocus={true}
            className="input__HtmlInput-rcagn3-0 iMczqN"
            name="reason"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Description"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="refund__ButtonContainer-a7ak7z-2 ldTkQC"
  >
    <button
      className="secondary__Button-j7mjyk-0 kgyVVR"
      disabled={false}
      onClick={null}
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      className="primary__Button-sc-1lwi4g7-0 loading__ExtendedButton-h7d8sr-0 dvYQJY"
      disabled={false}
      label="Refund the Transaction"
      onClick={null}
      type="submit"
    >
      <span>
        Refund the Transaction
      </span>
    </button>
  </div>
</form>
`;

exports[`RefundForm Should use loading status 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="refund__ContentContainer-a7ak7z-3 icpIBn"
  >
    <div
      className="refund__FormItemContainer-a7ak7z-1 iBWkWY"
    >
      <span
        className="refund__Value-a7ak7z-0 iETVAo"
      >
        <label
          className="radio__Label-sc-1hx8hak-0 kZePut"
          label="Partial"
        >
          Partial
          <input
            checked={false}
            name="refundType"
            onChange={[Function]}
            type="radio"
            value="partial"
          />
          <span />
        </label>
        <label
          className="radio__Label-sc-1hx8hak-0 kZePut"
          label="Full"
        >
          Full
          <input
            checked={true}
            name="refundType"
            onChange={[Function]}
            type="radio"
            value="full"
          />
          <span />
        </label>
      </span>
    </div>
    <div
      className="field_wrapper-n32k2e-0 irXinp"
      onClick={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <div
        style={
          Object {
            "minWidth": 100,
          }
        }
      >
        <h3
          className="h3-sc-1przhcr-0 bqBPGG"
        >
          Refund
        </h3>
      </div>
      <div
        className="input__InputErrorWrapper-sc-1dz7i8-0 cAUOWl"
      >
        <div
          className="input_wrapper-sc-15hmhfy-0 duFQAu"
          style={
            Object {
              "marginLeft": "16px",
              "width": "100%",
            }
          }
        >
          <input
            autoFocus={false}
            className="input_currency__HtmlInput-obstnl-0 cVRLQV"
            disabled={true}
            name="refund"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            placeholder="Refund"
            type="text"
            value="€12,345.00"
          />
        </div>
      </div>
    </div>
    <div
      className="field_wrapper-n32k2e-0 irXinp"
      onClick={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <div
        style={
          Object {
            "minWidth": 100,
          }
        }
      >
        <h3
          className="h3-sc-1przhcr-0 bqBPGG"
        >
          Description
        </h3>
      </div>
      <div
        className="input__InputErrorWrapper-sc-1dz7i8-0 cAUOWl"
      >
        <div
          className="input_wrapper-sc-15hmhfy-0 hwqBS"
          style={
            Object {
              "marginLeft": "16px",
              "width": "100%",
            }
          }
        >
          <input
            autoFocus={true}
            className="input__HtmlInput-rcagn3-0 iMczqN"
            name="reason"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Description"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="refund__ButtonContainer-a7ak7z-2 ldTkQC"
  >
    <button
      className="secondary__Button-j7mjyk-0 hSUMXm"
      disabled={true}
      onClick={null}
      type="button"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      className="primary__Button-sc-1lwi4g7-0 loading__ExtendedButton-h7d8sr-0 ffbUOi"
      disabled={true}
      label="Refund the Transaction"
      onClick={null}
      type="submit"
    >
      <svg
        height={24}
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 100 100"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          transform="rotate(0 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.9375s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(22.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.875s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(45 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.8125s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(67.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.75s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(90 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.6875s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(112.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.625s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(135 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.5625s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(157.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.5s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(180 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.4375s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(202.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.375s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(225 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.3125s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(247.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.25s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(270 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.1875s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(292.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.125s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(315 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="-0.0625s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
        <g
          transform="rotate(337.5 50 50)"
        >
          <rect
            fill="#ffffff"
            height="14"
            rx="2"
            ry="2"
            width="6"
            x="47"
            y="18"
          >
            <animate
              attributeName="opacity"
              begin="0s"
              dur="1s"
              repeatCount="indefinite"
              values="1;0"
            />
          </rect>
        </g>
      </svg>
    </button>
  </div>
</form>
`;
